src/rust/catnap/socket.rs

```diff
@@ -331,14 +181,235 @@ impl Socket {
     pub fn is_connecting(&self) -> bool {
         self.state == SocketState::Connecting
     }
+
+    /// Given the current state and the operation being executed, this function returns the next state on success and
+    fn get_transition(&self, op: SocketOp) -> Result<SocketState, Fail> {
```

Comment:
This function actually returns the next state. Would it be more appropriate to call it `get_next_state()` instead?